engine/CMakeLists.txt

```diff
@@ -636,25 +644,37 @@ IF (ENABLE_PIE)
   LINK_LIBRARIES("-pie")
 ELSE (ENABLE_PIE)
   MESSAGE("++ Disabling Position Independent Executables/Shared Libraries (Recommended on Ubuntu/Mint)")
-  ADD_COMPILE_OPTIONS("-no-pie")
-  LINK_LIBRARIES("-no-pie")
+  IF (NOT MSVC)
+    ADD_COMPILE_OPTIONS("-no-pie")
+    LINK_LIBRARIES("-no-pie")
+  ENDIF (NOT MSVC)
 ENDIF (ENABLE_PIE)
 
+IF (MSVC)
+    SET(COMPILER_SPECIFIC_FLAGS "/FI\"${Vega_Strike_BINARY_DIR}\"/config.h  -D__x86_64__ /Z7")
+    set(TOOLCHAIN_SPECIFIC_LINK_FLAGS "/DEBUG:FULL")
+ELSE (MSVC)
+    SET(COMPILER_SPECIFIC_FLAGS "-include config.h -pipe -Og -g3 -Wall -std=c++11 -fvisibility=hidden")
```

Comment:
`-Og` and `-g3` are, technically, specific not only to the gcc compiler, but to the Debug configuration as well. This will do for now, I suppose, but we'll want to fix this up at some point.

Reply:
`-0g` and `-g3` enable us to do release with debug info ;)